### ISSUE TITLE
[Testerina] Legacy function mocking throws java.lang.ClassNotFoundException

### ISSUE THREAD

**Issue by Ibaqu** [ballerina-platform/ballerina-lang#37007]
**Description:**

When using the legacy function mocking, the feature fails with a ClassNotFoundException. However, when used along side with the new function mocking, the issue does not arise. 

There seems to be a problem when loading classes for the legacy function mocking when not used with any other mocking feature. 

A temporary workaround to this problem if you need to use legacy function mocking is to make use of the new function mocking as well in the same file so that the classes get loaded properly. 

**Steps to reproduce:**
Simply use the legacy function mocking

```
test:Mock {
    functionName : <functionName>
}
public function functionNameMock() {
    // ... mock implementation
}
```

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

**Comment by Ibaqu** [comment]
Must test fix with the following scenario : https://github.com/ballerina-platform/ballerina-lang/issues/36585
Seems to be the same case

**Comment by gabilang** [comment]
The fix has been added to `2201.1.x` and `2201.2.x`. Hence closing.
